addons/godot-xr-tools/staging/scene_base.gd

```diff
@@ -60,28 +64,74 @@ func center_player_on(p_transform : Transform3D):
 	# And now update our origin point
 	$XROrigin3D.global_transform = (p_transform * transform.inverse()).orthonormalized()
 
-func scene_loaded():
+
+## This method is called when the scene is loaded, but before it becomes visible.
+##
+## The [param user_data] parameter is an optional parameter passed in when the
+## scene is loaded - usually from the previous scene. By default the
+## user_data can be a spawn-point node-name, Vector3, Transform3D, or
+## null to spawn at the scenes XROrigin3D location.
+##
+## Advanced scene-transition functionality can be implemented by overriding this
+## method and calling the super() with any desired spawn transform. This could
+## come from a field of an advanced user_data class-object, or from a game-state
+## singleton.
+func scene_loaded(user_data = null):
 	# Called after scene is loaded
 
 	# Make sure our camera becomes the current camera
 	$XROrigin3D/XRCamera3D.current = true
 	$XROrigin3D.current = true
 
-	# Center our player on our origin point
-	# Note, this means you can place the XROrigin3D point in the start
-	# position where you want the player to spawn, even if the player is
-	# physically halfway across the room.
-	center_player_on($XROrigin3D.global_transform)
+	# Inspect standard user data types to override the players starting location
+	# in a scene. Overriding this behavior can be achieved by overriding this
+	# function and calling super() with:
+	# - null to use the standard XROrigin3D location
+	# - string name of a Node3D to spawn at
+	# - Vector3 to spawn at
+	# - Transform3D to spawn at
+	var spawn_position : Transform3D = $XROrigin3D.global_transform
+	match typeof(user_data):
```

Comment:
I would add a `TYPE_OBJECT` in there as well, and check for say a method `get_spawn_position`, this way you could parse a "game data object" around that contains persistent game data from scene to scene. 

(or maybe we create a base class for this in XR Tools?)